src/main/scala/com/snowplowanalytics/indicative/Transformer.scala

```diff
@@ -81,18 +108,61 @@ object Transformer {
   private def constructIndicativeJson(eventName: String,
                                       uniqueId: String,
                                       properties: Map[String, Json],
-                                      eventTime: Long): JsonObject =
-    JsonObject(
+                                      eventTime: Long): Json =
+    Json.obj(
       "eventName"     -> Json.fromString(eventName),
       "eventUniqueId" -> Json.fromString(uniqueId),
       "eventTime"     -> Json.fromLong(eventTime),
       "properties"    -> Json.fromFields(properties)
     )
 
-  def constructBatchEvent(apiKey: String, events: List[JsonObject]): JsonObject =
-    JsonObject(
+  /**
+   * Construct batches of events which respect both the batch size limit (100) and the payload size
+   * limit (1Mb).
+   * @param apiKey the api key to fill the jsons with
+   * @param events the events to form batches from
+   * @param maxBatchSize maxmimum number of elements which can be in a batch
+   * @param maxBytesSize maximum size in bytes of a payload
+   * @return (a series of payloads ready to be sent, events that were too big to be sent (>=1Mb on
+   * their own))
+   */
+  def constructBatchesOfEvents(
```

Comment:
I feel that if we generalize it to

```
def constructBatch[A](getSize: A => Int, items: List[A], maxItems: Int, maxSize: Int): (List[A], List[A])
```

It will be easier to test and write properly.